pkg/proxy/handler/detected_fields.go

```diff
@@ -0,0 +1,297 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"sort"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/paulojmdias/lokxy/pkg/o11y/metrics"
+	traces "github.com/paulojmdias/lokxy/pkg/o11y/tracing"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// ===================== Common output types =====================
+
+// DetectedFieldOut is the unified output for /detected_fields
+type DetectedFieldOut struct {
+	Label       string   `json:"label"`
+	Type        string   `json:"type,omitempty"`
+	Cardinality int      `json:"cardinality"`
+	Parsers     []string `json:"parsers,omitempty"`
+}
+
+// LokiDetectedFieldsOut mirrors Loki's modern response (fields + optional limit).
+type LokiDetectedFieldsOut struct {
+	Fields []DetectedFieldOut `json:"fields"`
+	Limit  *int               `json:"limit,omitempty"`
+}
+
+// DetectedFieldValue represents one field value and its count
+type DetectedFieldValue struct {
+	Value string `json:"value"`
+	Count int    `json:"count"`
+}
+
+// LokiDetectedFieldValuesResponse represents detected_field/{name}/values response
+// We keep "field" for output to be stable w/ your router param, but accept upstream "label".
+type LokiDetectedFieldValuesResponse struct {
+	Field  string               `json:"field"`
+	Values []DetectedFieldValue `json:"values"`
+}
+
+// ===================== Input variants we accept =====================
+
+// Variant A (your cluster shows this)
+type detectedFieldsInA struct {
+	Fields []struct {
+		Label       string   `json:"label"`
+		Type        string   `json:"type"`
+		Cardinality int      `json:"cardinality"`
+		Parsers     []string `json:"parsers"`
+	} `json:"fields"`
+	Limit *int `json:"limit,omitempty"`
+}
+
+// Variant B (alternative/older shape)
+type detectedFieldsInB struct {
+	DetectedFields []struct {
+		// Some builds say "field", others might say "label"
+		Field       string `json:"field"`
+		Label       string `json:"label"`
+		Cardinality int    `json:"cardinality"`
+	} `json:"detectedFields"`
+}
+
+// Values input variant(s)
+type detectedFieldValuesIn struct {
+	// upstream may return "field" or "label"—we read either
+	Field  string `json:"field"`
+	Label  string `json:"label"`
+	Values []struct {
+		Value string `json:"value"`
+		Count int    `json:"count"`
+	} `json:"values"`
+}
+
+// dfAcc accumulates cardinality, first non-empty type, and a set of parsers.
+type dfAcc struct {
```

Comment:
a bit of a nit, but `dfAcc` sounds a bit confusing, I know go favors short names rather than long as per https://go.dev/wiki/CodeReviewComments#variable-names, but maybe change for something like fieldAgg and also change card to cardinality, for clarity?
```
type fieldAgg struct {
    cardinality int
    typ         string
    parsers     map[string]struct{}
}
```